docker/local/Dockerfile

```diff
@@ -0,0 +1,90 @@
+# Starts several services in a single container for local use
+# - Clickhouse
+# - Mongo
+# - Otel Collector (otelcol)
+# - Ingestor (Vector)
+# - API (Node)
+# - Aggregator (Node)
+# - App (Frontend)
+# - Redis Cache
+
+# TODO:
+# - Customize ports, need to set up env vars or relax CORS and other port requirements
+# - Have otel collector listen to a directory users can mount logs into
+# - Allow persisting settings on disk
+# - Limiting persisted data with some auto rotation
+
+FROM clickhouse/clickhouse-server:23.11.1-alpine AS base
+
+# ===
+# === Install Deps
+# ===
+
+# == Install Otel Collector Deps ==
+EXPOSE 1888 4317 4318 55679 13133
+
+RUN apk update
+RUN apk add wget shadow
+RUN wget https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/v0.90.1/otelcol-contrib_0.90.1_linux_arm64.apk
+RUN apk add --allow-untrusted otelcol-contrib_0.90.1_linux_arm64.apk
+
+# == Install Node Deps ==
+RUN apk add nodejs npm yarn
+
+# == Install Vector Deps ==
+RUN apk add curl
+RUN curl --proto '=https' --tlsv1.2 -sSfL https://sh.vector.dev | bash -s -- -y
```

Comment:
we also want to specify vector version (https://github.com/hyperdxio/hyperdx/blob/843cfa3ba8c7ae0f1104f880c77f3c5677591206/.github/workflows/main.yml#L28)